doc/code/qml_debugging.rst

```diff
@@ -0,0 +1,222 @@
+qml.debugging
+=============
+
+.. automodapi:: pennylane.debugging
+    :no-heading:
+    :no-inherited-members:
+    :skip: PLDB
+    :skip: pldb_device_manager
```

Comment:
Is it intended for snapshots to show up here? :thinking: 
Overall, it might make sense. Just curious.

Reply:
Though `snapshots` is also a top-level accessible function but it's showing up as `qml.debugging.snapshots`

Reply:
It was intended on purpose to place snapshots here. Originally, the snapshots logic and the debugger were both located in the same python file.